tools/monorepo-merge/src/commands/merge/index.ts

```diff
@@ -0,0 +1,205 @@
+	};
+
+	/**
+	 * This method is called to execute the command.
+	 */
+	async run(): Promise< void > {
+		const { args, flags } = await this.parse( Merge );
+
+		await this.checkDependencies();
+		await this.validateArgs( args.source, args.destination );
+
+		let confirmation = await CliUx.ux.confirm( 'WARNING: This command will DESTROY the history of your current branch. Are you sure you want to proceed? (y/n)' );
+		if ( ! confirmation ) {
+			this.exit( 0 );
+		}
+
+		const repositoryPath = await this.cloneRepository( args.source );
+		await this.alterRepositoryHistory( args.source, repositoryPath, args.destination );
+
+		confirmation = await CliUx.ux.confirm( 'Are you ready to merge ' + args.source + ' from ' + repositoryPath + '? (y/n)' );
+		if ( ! confirmation ) {
+			// Remove the repository we've cloned.
+			try {
+				await exec( 'rm -rf ' + repositoryPath );
+			} catch {}
+			
+			this.exit( 0 );
+		}
+		
+		await this.mergeRepository( args.source, repositoryPath, flags.branch );
+
+		this.log( 'Successfully merged ' + args.source + ' into ' + args.destination );
+	}
+
+	/**
+	 * Checks to make sure that all of the necessary dependencies to run the script are installed.
+	 */
+	private async checkDependencies(): Promise< void > {
+		try {
+			await exec( 'git --version' );
+		} catch {
+			this.error( '"git" must be installed' );
+		}
+
+		try {
+			await exec( 'git-filter-repo --version' );
+		} catch {
+			this.error( '"git-filter-repo" must be installed' );
+		}
+	}
+
+	/**
+	 * Validates all of the arguments to make sure they're compatible with the command.
+	 * 
+	 * @param {string} source The GitHub repository we are merging.
+	 * @param {string} destination The local path we're merging into.
+	 */
+	private async validateArgs( source: string, destination: string ): Promise< void > {
+		// We only support pulling from GitHub so the format needs to match that.
+		if ( ! source.match( /^[a-z0-9\-]+\/[a-z0-9\-]+$/ ) ) {
+			this.error(
+				'The "source" argument must be in "organization/repository" format'
+			);
+		}
+		
+		// We can't merge into a directory that already exists.
+		let exists = false;
+		try {
+			await access( join( MONOREPO_ROOT, destination ) );
+			exists = true;
+		} catch (err: any) {
+			exists = false;
+		}
+
+		if ( exists ) {
+			this.error('The "destination" argument points to a directory that already exists');
+		}
+	}
+
+	/**
+	 * Clones a repository from GitHub into a temporary directory and returns the path.
+	 * 
+	 * @param {string} source The GitHub repository we want to clone.
+	 */
+	private async cloneRepository(source: string): Promise< string > {
+		// Show progress for the cloning.
+		const gitPath = 'https://github.com/' + source;
+		CliUx.ux.action.start( 'Cloning from ' + gitPath );
+
+		// We need a fresh directory to clone the source into.
+		const cloneDir = join( tmpdir(), 'monorepo-merge', source );
+		try {
+			await access( cloneDir );
+			await exec( 'rm -rf ' + cloneDir );
```

Comment:
I don't think that `tmpdir()` is ever likely to return a pathname with spaces or other characters that need escaping, but if it does, this will likely break.